tutorial/tests/testsuite.py

```diff
@@ -23,20 +29,32 @@ def _name_from_ipynbname() -> str | None:
 
 
 def _name_from_globals(globals_dict: Dict) -> str | None:
-    module_path = globals_dict.get('__vsc_ipynb_file__') if globals_dict else None
+    """Find the name of the test module from the globals dictionary if working in VSCode"""
+    module_path = globals_dict.get("__vsc_ipynb_file__") if globals_dict else None
     return pathlib.Path(module_path).stem if module_path else None
 
 
 def get_module_name(line: str, globals_dict: Dict = None) -> str:
     """Fetch the test module name"""
-    module_name = _name_from_line(line) or _name_from_ipynbname() or _name_from_globals(globals_dict)
+    module_name = (
+        _name_from_line(line)
+        or _name_from_ipynbname()
+        or _name_from_globals(globals_dict)
+    )
 
     if not module_name:
-        raise RuntimeError("Test module is undefined. Did you provide an argument to %%ipytest?")
+        raise RuntimeError(
+            "Test module is undefined. Did you provide an argument to %%ipytest?"
+        )
 
     return module_name
 
 
+def find_solution(ns: Dict[str, Callable], postfix: str) -> Optional[Callable]:
+    """Given a namespace, finds a solution matching a given pattern."""
+    return [f for k, f in ns if k.endswith(postfix)][0]
```

Comment:
Are you sure it matches all solution patterns?  It only looks for files that end with `postfix`, what happens if we have for example "edoardo_exercise3"? Anyway this function is unused, so we can remove it